contracts/test/strategies/aerodrome-amo.base.fork-test.js

```diff
@@ -640,70 +704,68 @@ describe("ForkTest: Aerodrome AMO Strategy (Base)", function () {
 
       await expect(
         rebalance(
-          oethUnits("0.01"),
+          oethUnits("1000000000"),
           true, // _swapWETH
           oethUnits("0.009")
         )
       ).to.be.revertedWith("NotEnoughWethForSwap");
     });
 
     it("Should revert when pool rebalance is off target", async () => {
+      const { value, direction } = await quoteAmountToSwapBeforeRebalance();
+
       await expect(
-        rebalance(
-          oethUnits("0.001"),
-          true, // _swapWETH
-          oethUnits("0.0009")
-        )
+        rebalance(value.mul("200").div("100"), direction, 0)
```

Comment:
we can change this by fetching the swap amount by setting the price of the say tick 1 and tick -2 from the quoter

Reply:
I'm sorry, I didn't get what do you mean. You could explain it another way please?

Reply:
Sorry, for a bad initial comment :)

So what this test is doing is trying to trigger an error where too much funds would be swapped and the result would be a price out of tick. The way it is currently done is getting amount to swap to reach an ok target balance and multiplying the value by 2. While that might work right now I am afraid it doesn't make the test super robust, since there are probably base mainnet values where where this test won't trigger a fail. 

Now ignore my initial comment. 

I think the goal should be to get values in the test that will be always out of bounds no matter what the state of the mainnet base is.
You could probably achieve that by calling: `quoteAmountToSwapBeforeRebalance(0.98e18, 1e18)` and then multiplying that value by 2. Of if you can think of a more robust method yourself might be even better? 

Ideally we would set a price out of bounds. Say they price of the pool at ticker 1 (which is out of bounds) and get the value we need to swap for to reach that ticker when rebalancing.

Reply:
Thanks!
I've applied some changes, let me know what do you think of it ;)

> You could probably achieve that by calling: quoteAmountToSwapBeforeRebalance(0.98e18, 1e18) and then multiplying that value by 2.

I didn't multiplied it by 2, I just used the value obtained, otherwise I'll have push the price outside of bounds.
I used 0.90/0.92 for share instead of 0.98/1 due to this. 
https://github.com/OriginProtocol/origin-dollar/blob/944bb283ffa4204e41b05251173a99d11ff74ed4/contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol#L295

Reply:
Those changes look cool thanks, good thinking on how to bring the swap liquidity to the edge of the tick and not go over. 

My previous comment would actually apply better for testing this test `Should be reverted trying to rebalance and we are not in the correct tick` that is failing on [fork tests](https://github.com/OriginProtocol/origin-dollar/actions/runs/10885062587/job/30201758561). Which should trigger when the rebalance crosses the tick boundary. 
Would you mind adding the tick boundary tests for both directions please (crossing -1 tick going into -2 one. And crossing 0 tick going into tick 1). There are also a couple of other tests [that are failing](https://github.com/OriginProtocol/origin-dollar/actions/runs/10885062587/job/30201758561). 
But I think the approach is solid now 💪

Reply:
I've fixed the failing test (due to the new timelock mostly). 

I've preferred to push the price at tick -2 and then try to rebalance without swapping to force reverting on `OutsideExpectedTickRange` instead of `PoolRebalanceOutOfBounds` that will occur if I were modifying WethShareStart/End. 

So there are now two tests:
- when the pool is at tick -2 and we try to rebalance with `swappingAmount == 0`.
- when the pool is at tick 1 and we try to rebalance with `swappingAmount == 0`.

Let me know if this is ok for you, or if you prefer another approach 😉